File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/repositories/LedgerTxnRepository.java
package com.decrux.pocketr.api.repositories;

import com.decrux.pocketr.api.entities.db.ledger.LedgerTxn;
import java.util.List;
import java.util.UUID;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
import org.springframework.stereotype.Repository;

@Repository
public interface LedgerTxnRepository extends JpaRepository<LedgerTxn, UUID>, JpaSpecificationExecutor<LedgerTxn> {

    @EntityGraph(attributePaths = {"splits", "splits.account", "splits.categoryTag", "createdBy"})
    @Override
    List<LedgerTxn> findAll(Specification<LedgerTxn> spec);

    @EntityGraph(attributePaths = {"splits", "splits.account", "splits.categoryTag", "createdBy"})
    @Override
    Page<LedgerTxn> findAll(Specification<LedgerTxn> spec, Pageable pageable);
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr_api/repositories/LedgerTxnRepository.kt
package com.decrux.pocketr_api.repositories

import com.decrux.pocketr_api.entities.db.ledger.LedgerTxn
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.domain.Specification
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.jpa.repository.JpaSpecificationExecutor
import org.springframework.stereotype.Repository
import java.util.UUID

@Repository
interface LedgerTxnRepository : JpaRepository<LedgerTxn, UUID>, JpaSpecificationExecutor<LedgerTxn> {

    @EntityGraph(attributePaths = ["splits", "splits.account", "splits.categoryTag", "createdBy"])
    override fun findAll(spec: Specification<LedgerTxn>): List<LedgerTxn>

    @EntityGraph(attributePaths = ["splits", "splits.account", "splits.categoryTag", "createdBy"])
    override fun findAll(spec: Specification<LedgerTxn>, pageable: Pageable): Page<LedgerTxn>
}
